src/hotspot/share/gc/g1/heapRegionManager.cpp

```diff
@@ -567,10 +568,12 @@ bool HeapRegionManager::allocate_containing_regions(MemRegion range, size_t* com
     if (!curr_region->is_free()) {
```

Comment:
What happens with regions committed before this false return?

Reply:
I filed https://bugs.openjdk.org/browse/JDK-8309117.

Reply:
Actually I checked/tested again, and the result is that these unnecessarily committed regions will stay around as free regions. No other error or something. I adapted the CR.